### PR TITLE
DMWaveX model for Fourier representation of DM noise

### PR DESCRIPTION
**Pull request by abhisrkckl**
This is a clone of the WaveX model for DM variations.

### CODE REVIEW

**Comment by abhisrkckl** [comment]
Here are some test results:

Before applying DMWaveX: (reduced chi2 ~ 30)
image

After applying DMWaveX with 3 harmonics (reduced chisq ~ 9)
image

After applying DMWaveX with 9 harmonics (reduced chisq ~ 7.75)
image

DM time series reconstruction (9 harmonics)
image

File: src/pint/utils.py [hunk 1]
```diff
@@ -1366,6 +1366,86 @@ def wavex_setup(model, T_span, freqs=None, n_freqs=None):
     return model.components["WaveX"].get_indices()
 
 
+def dmwavex_setup(model, T_span, freqs=None, n_freqs=None):
+    """
+    Set-up a DMWaveX model based on either an array of user-provided frequencies or the wave number
+    frequency calculation. Sine and Cosine amplitudes are initially set to zero
+
+    User specifies T_span and either freqs or n_freqs. This function assumes that the timing model does not already
+    have any WaveX components. See add_dmwavex_component() or add_dmwavex_components() to add components
```
Comment (dlakaplan) [inline src/pint/utils.py:1375]: "any WaveX components": should be "DMWaveX"?

**Comment by dlakaplan** [comment]
Looks fine.  Do we need any utility to extract the DM time series (e.g., to convert to DMX) or is the `dmwavex_dm` method sufficient?

**Comment by abhisrkckl** [comment]
I think `dmwavex_dm` should be enough. Also, I think most users will only need `TimingModel.total_dm` normally.

File: src/pint/utils.py (hunk 1 above)
Comment (abhisrkckl) [reply src/pint/utils.py]: changed.

File: src/pint/utils.py [hunk 2]
```diff
@@ -1366,6 +1366,86 @@ def wavex_setup(model, T_span, freqs=None, n_freqs=None):
...
+    nyqist_freq = 1.0 / (2.0 * T_span)
+    if freqs is not None:
+        if isinstance(freqs, u.quantity.Quantity):
+            freqs.to(u.d**-1)
+        else:
+            freqs *= u.d**-1
+        if len(freqs) == 1:
+            model.DMWXFREQ_0001.quantity = freqs
+        else:
+            np.array(freqs)
```
Comment (dlakaplan) [inline src/pint/utils.py:1428]: should this be assigned to `freqs`?

Comment (abhisrkckl) [reply src/pint/utils.py]: Corrected,

Comment (abhisrkckl) [reply src/pint/utils.py]: the same mistake was in wavex also. I have corrected that as well.

**Pull request merged**